### DISCUSSION TITLE
Efficiently use drf_social_oauth2 with simple_jwt

### DISCUSSION THREAD

**Discussion by codewitgabi** [wagnerdelima/drf-social-oauth2#227] [Ideas]
So I was having issues using drf_social_oauth2 and simplejwt in my django application. I was able to find a way to use `Bearer` for oauth and then `JWT` for normal auth. But then of course, that does not make much sense so what I did was to modify drf_social_oauth2's ConvertTokenView in my application cause I also needed to pass custom data in my response and then from there, I was able to use do this;

```python
body = json_loads(body)

if "error" in body:
    return Response(data=body, status=status)

token = body.get("access_token")
user = User.objects.filter(oauth2_provider_accesstoken__token=token)[0]

refresh = RefreshToken.for_user(user)
access_token = str(refresh.access_token)
decoded_token = AccessToken(access_token)
expiration_time = datetime.fromtimestamp(decoded_token["exp"]) 
```

And then from here, I did this in my settings

```python
# before

REST_FRAMEWORK = {
    "DEFAULT_AUTHENTICATION_CLASSES": (
        "rest_framework_simplejwt.authentication.JWTAuthentication",
        "oauth2_provider.contrib.rest_framework.OAuth2Authentication",
        "drf_social_oauth2.authentication.SocialAuthentication",
    ),
}
```

```python
# after

REST_FRAMEWORK = {
    "DEFAULT_AUTHENTICATION_CLASSES": (
        "rest_framework_simplejwt.authentication.JWTAuthentication",
    ),
}
````

This way, it still uses only simple jwt. What do you guys think?

**Comment by samuelegin** [comment]
This is amazing keep up the good work

**Comment by izzyjosh** [comment]
Looking forward to implementing this
